stetho/src/main/java/com/facebook/stetho/inspector/elements/Descriptor.java

```diff
@@ -2,24 +2,39 @@
 
 package com.facebook.stetho.inspector.elements;
 
+import com.facebook.stetho.common.Util;
+
 import javax.annotation.Nullable;
 
 public abstract class Descriptor implements NodeDescriptor {
-
-  private Listener mListener;
+  private Host mHost;
 
   protected Descriptor() {
   }
 
-  void setListener(Listener listener) {
-    mListener = listener;
+  // This is called by DescriptorMap to hook us up to our Host (probably the DOMProvider).
+  // This method is idempotent in the sense that once you call it with a specific
+  // reference you must either 1) never call it again, or 2) call it again with that
+  // same reference.
+  void setHost(Host host) {
```

Comment:
Yes, I think it's as simple as that, though I don't have the full context like you do.